train.py

```diff
@@ -8,9 +8,15 @@
 import os
 import time
 from datetime import timedelta
+from functools import partial
+
+from typing import List, Optional, Set
 
 import torch
+
+from torch.distributed.device_mesh import DeviceMesh
 from torch.distributed.elastic.multiprocessing.errors import record
+from torch.nn.attention import sdpa_kernel, SDPBackend
```

Comment:
ditto: please make this import local, or e.g. just use `torch.nn.attention.sdpa_kernel` if it appears only once